File: Qb.Net/JoinCond.cs
using Viten.QueryBuilder.SqlOm;

namespace Viten.QueryBuilder
{
  /// <summary>Класс определения условий объединения</summary>
  public class JoinCond
  {
    internal JoinCondition Condition;

    internal JoinCond(string leftField, string rightField)
    {
      Condition = new JoinCondition(leftField, rightField);
    }

    /// <summary>Определение соединения по полям с одинаковыми именами</summary>
    public static JoinCond Fields(string field)
    {
      return JoinCond.Fields(field, field);
    }

    /// <summary>Определение соединения по полям с разными именами</summary>
    public static JoinCond Fields(string leftField, string rightField)
    {
      return new JoinCond(leftField, rightField);
    }

  }
}

File: Qb.Js/src/JoinCond.ts
import {JoinCondition} from "./JoinCondition";

export class JoinCond {
  private Condition: JoinCondition;
  constructor(leftField: string, rightField: string) {
    this.Condition = new JoinCondition(leftField, rightField);
  }

  static Fields(leftField: string, rightField: string): JoinCond;
  static Fields(field: string): JoinCond;
  static Fields(): JoinCond {
    var leftField: string;
    var rightField: string;
    if (arguments.length >= 1) {
      leftField = arguments[0];
      rightField = arguments[0];
    }
    if (arguments.length >= 2) {
      rightField = arguments[1];
    }
    return new JoinCond(leftField!, rightField!);
  }
}
